pkg/controller/shared/legacy_kube_informers.go

```diff
@@ -337,3 +339,115 @@ func (f *namespaceInformer) Lister() *cache.IndexerToNamespaceLister {
 	informer := f.Informer()
 	return &cache.IndexerToNamespaceLister{Indexer: informer.GetIndexer()}
 }
+
+type LimitRangeInformer interface {
+	Informer() framework.SharedIndexInformer
+	Indexer() cache.Indexer
+	Lister() StoreToLimitRangeLister
+}
+
+type limitRangeInformer struct {
+	*sharedInformerFactory
+}
+
+func (f *limitRangeInformer) Informer() framework.SharedIndexInformer {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+
+	informerObj := &kapi.LimitRange{}
+	informerType := reflect.TypeOf(informerObj)
+	informer, exists := f.informers[informerType]
+	if exists {
+		return informer
+	}
+
+	lw := f.customListerWatchers.GetListerWatcher(kapi.Resource("limitrange"))
+	if lw == nil {
+		lw = &cache.ListWatch{
+			ListFunc: func(options kapi.ListOptions) (runtime.Object, error) {
+				return f.kubeClient.LimitRanges(kapi.NamespaceAll).List(options)
+			},
+			WatchFunc: func(options kapi.ListOptions) (watch.Interface, error) {
+				return f.kubeClient.LimitRanges(kapi.NamespaceAll).Watch(options)
+			},
+		}
+	}
+
+	informer = framework.NewSharedIndexInformer(
+		lw,
+		informerObj,
+		f.defaultResync,
+		cache.Indexers{},
+	)
+	f.informers[informerType] = informer
+
+	return informer
+}
+
+func (f *limitRangeInformer) Indexer() cache.Indexer {
+	informer := f.Informer()
+	return informer.GetIndexer()
+}
+
+func (f *limitRangeInformer) Lister() StoreToLimitRangeLister {
+	informer := f.Informer()
+	return StoreToLimitRangeLister{Indexer: informer.GetIndexer()}
+}
+
+// StoreToReplicationControllerLister gives a store List and Exists methods. The store must contain only LimitRanges.
```

Comment:
The methods are fine since the resource is namespaced (see `storeLimitRangesNamespacer`). What is not fine is the place this code should live (`pkg/client/cache` instead of the shared informer library).